packages/fxa-settings/src/lib/glean/index.ts

```diff
@@ -204,13 +218,14 @@ export const GleanMetrics: Pick<
     }),
 };
 
-for (const [page, events] of Object.entries(eventsMap)) {
-  (GleanMetrics as GleanMetricsT)[page as EventMapKeys] = Object.entries(
-    events
-  ).reduce((acc: { [key: string]: PingFn }, [evt, name]) => {
-    acc[evt] = createEventFn(name);
-    return acc;
-  }, {});
+for (const [page, events] of Object.entries(eventsMap) as Entries<EventsMap>) {
+  (GleanMetrics as any)[page] = Object.entries(events).reduce(
```

Comment:
Not a fan of casting as `any`, but maybe this is a case to bend the rules.

Reply:
Heh, well I did spend some time on this but couldn't quite get it.  But the type is asserted before it's exported so to the callers it's still the correct type.